scripts/cmake/vcpkg_from_git.cmake

```
@@ -80,29 +80,29 @@ function(vcpkg_from_git)
     vcpkg_list(SET git_fetch_shallow_param --depth 1)
     vcpkg_list(SET extract_working_directory_param)
     set(git_working_directory "${DOWNLOADS}/git-tmp")
+
+    if(DEFINED arg_FETCH_REF)
+        set(ref_to_use "${arg_FETCH_REF}")
+        vcpkg_list(SET git_fetch_shallow_param)
+    else()
+        set(ref_to_use "${arg_REF}")
+    endif()
+    string(REPLACE "/" "_-" sanitized_ref "${arg_REF}")
+
     if(VCPKG_USE_HEAD_VERSION)
         if(DEFINED arg_HEAD_REF)
```

I think this logic would be simpler as:
```suggestion
    if(VCPKG_USE_HEAD_VERSION AND DEFINED arg_HEAD_REF)
       # head impl
    else()
       if(VCPKG_USE_HEAD_VERSION)
         message(STATUS "Package does not specify HEAD_REF. Falling back to non-HEAD version.")
       endif()
       # non-HEAD impl
   endif()
```

---

sure, makes sense